dipy/align/crosscorr.pyx

```diff
@@ -27,17 +28,115 @@ cdef enum:
+        the moving volume (notice that both images must already be in a common
+        reference domain, i.e. the same S, R, C)
+    static : array, shape (S, R, C)
+        the static volume, which also defines the reference registration domain
+    ss : int
+        first coordinate of the rectangular window to be updated
+    rr : int
+        second coordinate of the rectangular window to be updated
+    cc : int
+        third coordinate of the rectangular window to be updated
+    s: int
+        first coordinate of the voxel the local window should be updated with
+    r: int
+        second coordinate of the voxel the local window should be updated with
+    c: int
+        third coordinate of the voxel the local window should be updated with
+    operation : int, either -1, 0 or 1
+        indicates whether the factors of voxel (`s`, `r`, `c`) should be
+        added to (`operation`=1), subtracted from (`operation`=-1), or set as
+        (`operation`=0) the current factors for the rectangular window centered
+        at (`ss`, `rr`, `cc`).
+
+    """
+    cdef:
+        double sval
+        double mval
+    if s>=moving.shape[0] or r>=moving.shape[1] or c>=moving.shape[2]:
+        if operation == 0:
+            factors[ss, rr, cc, SI] = 0
+            factors[ss, rr, cc, SI2] = 0
+            factors[ss, rr, cc, SJ] = 0
+            factors[ss, rr, cc, SJ2] = 0
+            factors[ss, rr, cc, SIJ] = 0
+    else:
+        sval = static[s,r,c]
+        mval = moving[s,r,c]
+        if operation == 0:
+            factors[ss, rr, cc, SI] = sval
+            factors[ss, rr, cc, SI2] = sval*sval
+            factors[ss, rr, cc, SJ] = mval
+            factors[ss, rr, cc, SJ2] = mval*mval
+            factors[ss, rr, cc, SIJ] = sval*mval
+        elif operation == -1:
+            factors[ss, rr, cc, SI] -= sval
+            factors[ss, rr, cc, SI2] -= sval*sval
+            factors[ss, rr, cc, SJ] -= mval
+            factors[ss, rr, cc, SJ2] -= mval*mval
+            factors[ss, rr, cc, SIJ] -= sval*mval
+        elif operation == 1:
+            factors[ss, rr, cc, SI] += sval
+            factors[ss, rr, cc, SI2] += sval*sval
+            factors[ss, rr, cc, SJ] += mval
+            factors[ss, rr, cc, SJ2] += mval*mval
+            factors[ss, rr, cc, SIJ] += sval*mval
+
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+@cython.cdivision(True)
+def precompute_cc_factors_3d(floating[:, :, :] static,
+                             floating[:, :, :] moving,
+                             cnp.npy_intp radius, num_threads=None):
     r"""Precomputations to quickly compute the gradient of the CC Metric
 
     Pre-computes the separate terms of the cross correlation metric and image
     norms at each voxel considering a neighborhood of the given radius to
     efficiently compute the gradient of the metric with respect to the
-    deformation field [Avants08][Avants11]
+    deformation field [Ocegueda16][Avants08][Avants11].
```

Comment:
Thanks @arokem!, =)
I just changed the citations in `crosscorr.pyx`. Should I fix the format of all references in the `align` module on this PR, or make a separate one?